src/services/goToDefinition.ts

```diff
@@ -326,10 +326,41 @@ namespace ts.GoToDefinition {
                 sourceFile,
                 FindAllReferences.getContextNode(declaration)
             ),
-            isLocal: !checker.isDeclarationVisible(declaration)
+            isLocal: !isDefinitionVisible(checker, declaration)
         };
     }
 
+    function isDefinitionVisible(checker: TypeChecker, declaration: Declaration): boolean {
+        if (checker.isDeclarationVisible(declaration)) return true;
+        if (!declaration.parent) return false;
+
+        // Assignments to variable are visible if variable is visible
+        if (hasInitializer(declaration.parent) && declaration.parent.initializer === declaration) return isDefinitionVisible(checker, declaration.parent as Declaration);

```

Comment:
This is for cases like `const c = function() { ... }`, where `isDefinitionVisible` is called on the function expression?

Reply:
Maybe "assignments to variables" would be clearer as "variable initializers"?